packages/host/app/resources/auto-attached-card.ts

```diff
@@ -51,29 +58,50 @@ export class AutoAttachment extends Resource<Args> {
 
   private calculateAutoAttachments = restartableTask(
     async (
+      submode: Submode,
+      autoAttachedFileUrl: string | undefined,
+      playgroundPanelCardId: string | undefined,
       topMostStackItems: StackItem[],
       attachedCardIds: string[] | undefined,
       removedCardIds: string[] | undefined,
     ) => {
       this.cardIds.clear();
-      for (let item of topMostStackItems) {
-        if (!item.id) {
-          continue;
-        }
-        if (removedCardIds?.includes(item.id)) {
-          continue;
-        }
-        if (attachedCardIds?.includes(item.id)) {
-          continue;
-        }
-        let card = await this.store.get(item.id);
-        if (card && isCardInstance(card)) {
-          let realmURL = card[realmURLSymbol];
-          if (realmURL && item.id === `${realmURL.href}index`) {
+      if (submode === Submodes.Interact) {
+        for (let item of topMostStackItems) {
+          if (!item.id) {
             continue;
           }
+          if (removedCardIds?.includes(item.id)) {
+            continue;
+          }
+          if (attachedCardIds?.includes(item.id)) {
+            continue;
+          }
+          let card = await this.store.get(item.id);
+          if (card && isCardInstance(card)) {
+            let realmURL = card[realmURLSymbol];
+            if (realmURL && item.id === `${realmURL.href}index`) {
+              continue;
+            }
+          }
+          this.cardIds.add(item.id);
+        }
+      } else if (submode === Submodes.Code) {
```

Comment:
i'm questioning why you actually even care about the submode. cannot you not collapse these two different branches of the `if` into a single set of operations that works in either mode? like `autoAttachedFileUrl` should _always_ get the `.json` suffix treatment regardless of which submode you are in right? and why is  `playgroundPanelCardId` different from `attachedCardIds`, what makes its treatment so special, its just a card ID you want to attach, right?